### DISCUSSION TITLE
stable companion ovos package release

### DISCUSSION THREAD

**Discussion by JarbasAl** [OpenVoiceOS/ovos-buildroot#100] [General]
ovos-core `0.1.0` is around the corner! :tada:

we need a general review and new stable release in the companion packages so we can pin them

These will be considered the first stable releases of all these packages! :tada: 

- [ ] [ovos_utils](https://github.com/OpenVoiceOS/ovos_utils) `0.1.0`
- [ ] [ovos_workshop](https://github.com/OpenVoiceOS/OVOS-workshop)  `0.1.0`
- [ ] [ovos_plugin_manager](https://github.com/OpenVoiceOS/OVOS-plugin-manager) `0.1.0`
- [ ] [ovos_skills_manager](https://github.com/OpenVoiceOS/ovos_skill_manager) `0.1.0`

It will also be the first alpha release of PHAL  - PluginHardwareAbstractionLayer :tada:

- [x] [PHAL](https://github.com/OpenVoiceOS/ovos_PHAL) `0.0.1`
- [ ] [ovos-PHAL-plugin-mk1](https://github.com/OpenVoiceOS/ovos-PHAL-plugin-mk1) `0.0.1`
- [ ] [ovos-PHAL-plugin-connectivity-events](https://github.com/OpenVoiceOS/ovos-PHAL-plugin-connectivity-events)  `0.0.1`
- [ ] [ovos-PHAL-plugin-display-manager-ipc](https://github.com/OpenVoiceOS/ovos-PHAL-plugin-display-manager-ipc)  `0.0.1`
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-gui-network-client
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-balena-wifi
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-wifi-setup
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-network-manager
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-system
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-notification-widgets
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-mk2
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-alsa
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-dashboard
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-brightness-control-rpi
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-color-scheme-manager
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-configuration-provider
- [ ] https://github.com/OpenVoiceOS/ovos-PHAL-plugin-ipgeo

Repos with automated unit / build / publish / license tests

- [x] ovos_utils
- [x] ovos_workshop
- [x] ovos_plugin_manager
- [x] ovos_skills_manager

plugins with automated unit / build / publish / license tests

- [x] mimic
- [x] mimic2
- [x] precise
- [x] precise-lite
- [x] snowboy
- [x] ocp
- [x] vlc
- [x] simple audio

Issues to be solved before release:
- [x] https://github.com/OpenVoiceOS/ovos-core/issues/51
- [x] https://github.com/OpenVoiceOS/ovos_utils/issues/5
- [x] https://github.com/OpenVoiceOS/ovos_utils/pull/11
- [ ] https://github.com/OpenVoiceOS/ovos_skill_manager/issues/62
- [x] https://github.com/OpenVoiceOS/ovos_skill_manager/issues/82
- [ ] https://github.com/OpenVoiceOS/ovos_skill_manager/issues/84
- [ ] https://github.com/OpenVoiceOS/ovos_skill_manager/issues/83
- [x] https://github.com/OpenVoiceOS/ovos_skill_manager/pull/81
- [x] https://github.com/OpenVoiceOS/ovos_skill_manager/pull/79
- [x]  https://github.com/OpenVoiceOS/ovos_skill_manager/pull/76
- [x] https://github.com/OpenVoiceOS/OVOS-plugin-manager/pull/18
- [ ] https://github.com/OpenVoiceOS/OVOS-plugin-manager/pull/17
- [x] https://github.com/OpenVoiceOS/OVOS-plugin-manager/pull/6
- [ ] ...

**Comment by JarbasAl** [comment]
minimal unit tests that we can run on every PR moving forward should also be added before we call `0.1.0`for any of those packages

lets not worry about coverage just yet, i suggest we focus on having all the automation in place and add new tests over time

**Comment by NeonDaniel** [comment]
ovos_skills_manager has coverage/automation and should be ready with these open PR's merged:
https://github.com/OpenVoiceOS/ovos_skill_manager/pull/81
https://github.com/OpenVoiceOS/ovos_skill_manager/pull/79
https://github.com/OpenVoiceOS/ovos_skill_manager/pull/76

**Reply by JarbasAl** [reply]
added to initial post

**Reply by NeonDaniel** [reply]
I think we're ready for a new alpha release after approval/merge of https://github.com/OpenVoiceOS/ovos_skill_manager/pull/81

**Reply by NeonDaniel** [reply]
Do we want to knock out some of these "simple" issues too? Or wait for a patch release
https://github.com/OpenVoiceOS/ovos_skill_manager/issues/47
https://github.com/OpenVoiceOS/ovos_skill_manager/issues/38
https://github.com/OpenVoiceOS/ovos_skill_manager/issues/41
https://github.com/OpenVoiceOS/ovos_skill_manager/issues/74

**Reply by JarbasAl** [reply]
if we can fix them lets do it

**Reply by NeonDaniel** [reply]
Closed this one as later versions are not affected https://github.com/OpenVoiceOS/ovos_skill_manager/issues/82

**Reply by NeonDaniel** [reply]
Should we mark OSM as 0.1.0 so we can pin it as the first 'stable' version?

**Comment by NeonDaniel** [comment]
ovos-utils I think just needs these addressed:
https://github.com/OpenVoiceOS/ovos_utils/issues/5
https://github.com/OpenVoiceOS/ovos_utils/pull/11

But, this one might be ready to go too?
https://github.com/OpenVoiceOS/ovos_utils/pull/1

**Reply by NeonDaniel** [reply]
same as above, should we call the current "stable" 0.1.0?

**Comment by JarbasAl** [comment]
added PHAL repos to initial post

**Comment by JarbasAl** [comment]
lots of automations in the past weeks!

updated the list above, now need to update the issues...
